### PR TITLE
Fix dialyzer failures by removing skewed merkerl from proto

### PR DESCRIPTION
**Pull request by vihu**
- `blockchain_state_channel_v1:new/2` now returns a two-tuple
- Update tests
- Add new eunit
- Change validation logic for state channel close txn

### CODE REVIEW

File: src/ledger/v1/blockchain_ledger_snapshot_v1.erl [hunk 1]
```diff
@@ -222,11 +222,9 @@ serialize(Snapshot, BlocksP) ->
              BinSz:32/little-unsigned-integer, Bin/binary>>,
     {ok, Snap}.
 
-serialize_v1(Snapshot, BlocksP) ->
-    Snapshot1 = case BlocksP of
-                    blocks -> Snapshot;
-                    noblocks -> Snapshot#blockchain_snapshot_v1{blocks = []}
-                end,
+serialize_v1(Snapshot, noblocks) ->
```
Comment (mrallen1) [inline src/ledger/v1/blockchain_ledger_snapshot_v1.erl:225]: Got this error from dialyzer: 

```
===> Doing success typing analysis...
===> Resolving files...
===> Analyzing 96 files with "/home/mallen/github/helium/blockchain-core/_build/default/rebar3_22.3.4_plt"...

src/ledger/v1/blockchain_ledger_snapshot_v1.erl
 227: The pattern 'blocks' can never match the type 'noblocks'
```

Comment (vihu) [reply src/ledger/v1/blockchain_ledger_snapshot_v1.erl:225]: Right I fixed that in the latest commit, had to force push the branch so you may want to try: `git reset --hard origin/rg/fix-dialyzer`

File: rebar.config [hunk 2]
```diff
@@ -39,7 +39,7 @@
     {erlang_stats, ".*", {git, "https://github.com/helium/erlang-stats.git", {branch, "master"}}},
     {e2qc, ".*", {git, "https://github.com/project-fifo/e2qc", {branch, "master"}}},
     {vincenty, ".*", {git, "https://github.com/helium/vincenty", {branch, "master"}}},
-    {helium_proto, {git, "https://github.com/helium/proto.git", {branch, "master"}}},
+    {helium_proto, {git, "https://github.com/helium/proto.git", {branch, "rg/fix-dialyzer"}}},
```
Comment (madninja) [inline rebar.config:42]: hmm

File: src/state_channel/blockchain_state_channels_server.erl [hunk 3]
```diff
@@ -134,11 +135,11 @@ handle_cast({packet, SCPacket},
             %% Get raw packet
             Packet = blockchain_state_channel_packet_v1:packet(SCPacket),
             %% ActiveSCID should always be in our state_channels map
-            SC = maps:get(ActiveSCID, SCs),
+            {SC, Skewed} = maps:get(ActiveSCID, SCs),
```
Comment (madninja) [inline src/state_channel/blockchain_state_channels_server.erl:138]: Why is Skewed still getting referred to here? Name better?

File: src/state_channel/v1/blockchain_state_channel_v1.erl [hunk 4]
```diff
@@ -52,26 +51,26 @@ new(ID, Owner) ->
...
-        skewed=undefined,
         state=open,
         expire_at_block=0
     }.
 
 -spec new(ID :: binary(),
           Owner :: libp2p_crypto:pubkey_bin(),
           BlockHash :: binary(),
-          ExpireAtBlock :: pos_integer()) -> state_channel().
+          ExpireAtBlock :: pos_integer()) -> {state_channel(), skewed:skewed()}.
```
Comment (madninja) [inline src/state_channel/v1/blockchain_state_channel_v1.erl:61]: Skewed?!

**Comment by vihu** [comment]
@madninja This does _not_ remove the skewed merkle tree, it only stores it separately from the proto. We still require the skewed tree for packet tracking.

File: src/state_channel/v1/blockchain_state_channel_v1.erl (hunk 4 above)
Comment (vihu) [reply src/state_channel/v1/blockchain_state_channel_v1.erl:61]: Refer my general comment, this is required.

File: src/state_channel/blockchain_state_channels_server.erl (hunk 3 above)
Comment (madninja) [reply src/state_channel/blockchain_state_channels_server.erl:138]: I was wrong in assuming skewed merkle was getting removed.. All good

**Pull request merged**